Resolve deprecation warnings

MaartenBent opened the pull request:
Deprecation warnings in demos and utils, and a deprecation warning in the socket client example.

vadz commented on demos/bombs/bombs1.cpp:
```diff
@@ -36,10 +36,10 @@ void BombsCanvas::DrawField(wxDC *dc, int xc1, int yc1, int xc2, int yc2)
     wxColour wxFocused = wxTheColourDatabase->Find(wxT("GREY"));
 
-    wxPen *bluePen = wxThePenList->FindOrCreatePen(*wxBLUE, 1, wxSOLID);
+    wxPen *bluePen = wxThePenList->FindOrCreatePen(*wxBLUE, 1, wxPENSTYLE_SOLID);
```
We should just remove the `1, wxSOLID` part, it's the default, anyhow.

vadz commented on demos/bombs/bombs.h:
```diff
@@ -118,7 +118,7 @@ class BombsCanvas : public wxPanel
  */
 
-#define BOMBS_FONT wxFont(14, wxROMAN, wxNORMAL, wxNORMAL)
+#define BOMBS_FONT wxFont(14, wxFONTFAMILY_ROMAN, wxFONTSTYLE_NORMAL, wxFONTWEIGHT_NORMAL)
```
As long as we modernize this, let's use `wxFontInfo(14).Family(wxFONTFAMILY_ROMAN)` instead which is shorter and arguably more clear.

vadz commented on demos/forty/canvas.cpp:
```diff
@@ -42,9 +42,9 @@ FortyCanvas::FortyCanvas(wxWindow* parent, const wxPoint& pos, const wxSize& siz
 
 #ifdef __WXGTK__
-    m_font = wxTheFontList->FindOrCreateFont(12, wxROMAN, wxNORMAL, wxNORMAL);
+    m_font = wxTheFontList->FindOrCreateFont(12, wxFONTFAMILY_ROMAN, wxFONTSTYLE_NORMAL, wxFONTWEIGHT_NORMAL);
```
This can't be abbreviated currently, unfortunately, but maybe it'd be worth adding `FindOrCreateFont(const wxFontInfo&)` overload? Then we could do the same thing as for the bombs font above.

vadz commented on demos/fractal/fractal.cpp:
```diff
@@ -149,16 +149,16 @@ MyCanvas::MyCanvas(wxFrame *frame):
 {
     wxColour wxCol1(255,255,255);
-    SnowPen = wxPen(wxCol1, 2, wxSOLID);
+    SnowPen = wxPen(wxCol1, 2, wxPENSTYLE_SOLID);
```
Solid style is also the default here and could (and should IMO) be just removed.

vadz commented on samples/sockets/client.cpp:
```diff
@@ -594,7 +594,7 @@ void DoDownload(const wxString& urlname)
     // Try to get the input stream (connects to the given URL)
     wxLogMessage("Establishing connection to \"%s\"...", urlname);
-    const std::auto_ptr<wxInputStream> data(url.GetInputStream());
+    const wxScopedPtr<wxInputStream> data(url.GetInputStream());
```
We should include `wx/scopedptr.h` from here now (but, to compensate, we probably can remove the `#include <memory>`).

vadz reviewed (approved):
Thanks for the fixes and we can merge it as is, but if you could simplify/modernize things further, it would be even more welcome.

MaartenBent committed:
Replace deprecated std::auto_ptr with wxScopedPtr

MaartenBent committed:
Add FindOrCreateFont overload accepting wxFontInfo

MaartenBent committed:
Fix wxPen and wxBrush style deprecation warnings in utils

Fix a warning about hidden local variable in the helpview util.

MaartenBent committed:
Fix warnings and modernize demo code

Fix wxFont, wxPen and wxBrush style deprecation warnings.
Add wxFALLTHROUGH to resolve implicit fallthrough warnings in switch statements.
Fix build error when using long for GetTextExtent (with wxUSE_GRID 0).
Remove unused icon from fractal demo.

MaartenBent committed:
Fix reading poems from file in wxPoem demo

When using ftell to get the position in the file, it should be opened in binary mode.
Regenerate the poem index file with the correct start positions.
Use the wxICON macro for creating the icon, so .ico is used on Windows.

MaartenBent committed:
More use of wxFALLTHROUGH in MSW code and samples

MaartenBent committed:
Fix unused variable and parameter warnings in samples

These variables are only used in debug builds, causing warnings in release builds.

vadz commented:
Thanks for the updates!
